runhouse/rns/folders/folder.py

```diff
@@ -758,14 +761,17 @@ def rm(self, name, recursive: bool = True):
         except FileNotFoundError:
             pass
 
-    def put(self, contents, overwrite=False):
-        """Put given contents in folder. Contents must be one of the following:
-
-        - Dict with keys being the file names and values being the file-like objects to write
-
-        - Resource
-
-        - List of Resources.
+    def put(self, contents, overwrite=False, as_json=False):
+        """Put given contents in folder.
+
+        Args:
+            contents (Dict[str, Any] or Resource or List[Resource]): Contents to put in folder.
+                Must be a dict with keys being the file names (without full paths) and values being the file-like
+                objects to write, or a Resource object, or a list of Resources.
+            overwrite (bool): Whether to dump the file contents as json. By default expects data to be encoded.
+                Defaults to ``False``.
+            as_json (bool): Name of the Run to create. If ``True``, a Run will be created
+                Defaults to ``False``.
```

Comment:
1) I think this was copy pasted
2) I don't think we should handle serialization here, I think it's too automagic. Json serialization fails often and users will blame us, when really it's just debugging they need to do to make sure their object can be written down properly. If there's a big convenience to not needing to convert all the way to bytes before passing into .put, one thing we can do is support an argument like "write_fn" (which accepts a file and data) which can do the serialization, so the user can pass `write_fn=lambda f, data: json.dump(data, f)` which we then call inside put. This would mean the burden is still on them to handle the serialization, but it can also support any other serialization format. We don't need to accept args and kwargs into the write_fn because the user can just just functools `partial` to pre-pass those.

Reply:
Btw we should also accept "mode" as an arg into put, users want to be able to control that (just like we do here).